<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Imago</class>
 <widget class="QMainWindow" name="Imago">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1147</width>
    <height>798</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Imago</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QLabel" name="LabelInput">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>40</y>
      <width>421</width>
      <height>341</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="LabelOutput">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>40</y>
      <width>421</width>
      <height>341</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QListWidget" name="techniquesListWidget">
    <property name="geometry">
     <rect>
      <x>760</x>
      <y>411</y>
      <width>221</width>
      <height>261</height>
     </rect>
    </property>
   </widget>
   <widget class="QFrame" name="processFlowFrame">
    <property name="geometry">
     <rect>
      <x>760</x>
      <y>680</y>
      <width>221</width>
      <height>51</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Panel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Sunken</enum>
    </property>
    <widget class="QPushButton" name="moveProcessDown">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>10</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="shortcut">
      <string>Ctrl+Down</string>
     </property>
    </widget>
    <widget class="QPushButton" name="removeProcess">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>10</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="shortcut">
      <string>Backspace</string>
     </property>
    </widget>
    <widget class="QPushButton" name="moveProcessUp">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>10</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="shortcut">
      <string>Ctrl+Up</string>
     </property>
    </widget>
    <widget class="QPushButton" name="refreshProcess">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>10</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="whatsThis">
      <string/>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="shortcut">
      <string>Ctrl+Shift+R</string>
     </property>
     <property name="flat">
      <bool>false</bool>
     </property>
    </widget>
   </widget>
   <widget class="QTabWidget" name="tabProcesses">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>400</y>
      <width>561</width>
      <height>331</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>Basic</string>
     </attribute>
     <widget class="QGroupBox" name="gBoxFlip">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>0</y>
        <width>191</width>
        <height>111</height>
       </rect>
      </property>
      <property name="title">
       <string>Flip image</string>
      </property>
      <widget class="QRadioButton" name="flipHorizontal">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>102</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Horizontally</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="flipVertical">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>50</y>
         <width>102</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Vertically</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="flipBoth">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>70</y>
         <width>102</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Both</string>
       </property>
      </widget>
      <widget class="QPushButton" name="AddFlipImage">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>30</y>
         <width>21</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox">
      <property name="geometry">
       <rect>
        <x>280</x>
        <y>10</y>
        <width>191</width>
        <height>91</height>
       </rect>
      </property>
      <property name="title">
       <string>Salt and pepper noise</string>
      </property>
      <widget class="QSlider" name="sp_noiselevel">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>60</y>
         <width>160</width>
         <height>22</height>
        </rect>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
      <widget class="QPushButton" name="AddSaltAndPepper">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>30</y>
         <width>21</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="geometry">
       <rect>
        <x>280</x>
        <y>130</y>
        <width>191</width>
        <height>61</height>
       </rect>
      </property>
      <property name="title">
       <string>Histogram Equalization</string>
      </property>
      <widget class="QPushButton" name="AddEqualizeHistogram">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>30</y>
         <width>21</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>Filtering</string>
     </attribute>
     <widget class="QGroupBox" name="gBoxMorphology">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>10</y>
        <width>191</width>
        <height>111</height>
       </rect>
      </property>
      <property name="title">
       <string>Morphology</string>
      </property>
      <widget class="QPushButton" name="AddMorphologyOperation">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>30</y>
         <width>21</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
      <widget class="QSpinBox" name="MorphologySESize">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>70</y>
         <width>49</width>
         <height>24</height>
        </rect>
       </property>
      </widget>
      <widget class="QComboBox" name="MorphologySE">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>70</y>
         <width>111</width>
         <height>26</height>
        </rect>
       </property>
      </widget>
      <widget class="QComboBox" name="MorphologyOperation">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>111</width>
         <height>26</height>
        </rect>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="gBoxLowPassFilters">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>150</y>
        <width>191</width>
        <height>101</height>
       </rect>
      </property>
      <property name="title">
       <string>Low pass filters</string>
      </property>
      <widget class="QComboBox" name="LowPassFilter">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>60</y>
         <width>111</width>
         <height>26</height>
        </rect>
       </property>
      </widget>
      <widget class="QSpinBox" name="LowPassFilterSize">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>60</y>
         <width>49</width>
         <height>24</height>
        </rect>
       </property>
      </widget>
      <widget class="QPushButton" name="AddLowPassFilter">
       <property name="geometry">
        <rect>
         <x>163</x>
         <y>30</y>
         <width>21</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox_4">
      <property name="geometry">
       <rect>
        <x>280</x>
        <y>10</y>
        <width>181</width>
        <height>111</height>
       </rect>
      </property>
      <property name="title">
       <string>High pass filters</string>
      </property>
      <widget class="QComboBox" name="HighPassFilter">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>60</y>
         <width>111</width>
         <height>26</height>
        </rect>
       </property>
      </widget>
      <widget class="QPushButton" name="AddHighPassFilter">
       <property name="geometry">
        <rect>
         <x>150</x>
         <y>30</y>
         <width>21</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_3">
     <attribute name="title">
      <string>Shape</string>
     </attribute>
     <widget class="QGroupBox" name="groupBox_3">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>20</y>
        <width>201</width>
        <height>101</height>
       </rect>
      </property>
      <property name="title">
       <string>Canny edge detection</string>
      </property>
      <widget class="QSlider" name="sliderCanny">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>60</y>
         <width>160</width>
         <height>22</height>
        </rect>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
      <widget class="QPushButton" name="AddCanny">
       <property name="geometry">
        <rect>
         <x>170</x>
         <y>30</y>
         <width>21</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="gBoxHoughCircles">
      <property name="geometry">
       <rect>
        <x>290</x>
        <y>20</y>
        <width>181</width>
        <height>161</height>
       </rect>
      </property>
      <property name="title">
       <string>Hough circles</string>
      </property>
      <widget class="QPushButton" name="AddHoughCircles">
       <property name="geometry">
        <rect>
         <x>150</x>
         <y>30</y>
         <width>21</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
      <widget class="QSpinBox" name="HCminRad">
       <property name="geometry">
        <rect>
         <x>80</x>
         <y>120</y>
         <width>49</width>
         <height>24</height>
        </rect>
       </property>
      </widget>
      <widget class="QSpinBox" name="HCmaxRad">
       <property name="geometry">
        <rect>
         <x>140</x>
         <y>120</y>
         <width>49</width>
         <height>24</height>
        </rect>
       </property>
      </widget>
      <widget class="QSlider" name="sliderHCCanny">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>60</y>
         <width>160</width>
         <height>22</height>
        </rect>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
      <widget class="QSlider" name="sliderHCVotes">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>90</y>
         <width>160</width>
         <height>22</height>
        </rect>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="HCminDist">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>120</y>
         <width>69</width>
         <height>24</height>
        </rect>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="gBoxHoughLines">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>140</y>
        <width>191</width>
        <height>131</height>
       </rect>
      </property>
      <property name="title">
       <string>Hough lines</string>
      </property>
      <widget class="QPushButton" name="AddHoughLines">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>30</y>
         <width>21</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
      <widget class="QSlider" name="sliderHLthreshhold">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>100</y>
         <width>160</width>
         <height>22</height>
        </rect>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="HLrho">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>70</y>
         <width>69</width>
         <height>24</height>
        </rect>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="HLtheta">
       <property name="geometry">
        <rect>
         <x>100</x>
         <y>70</y>
         <width>69</width>
         <height>24</height>
        </rect>
       </property>
      </widget>
      <widget class="QComboBox" name="HoughLinesOperation">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>40</y>
         <width>111</width>
         <height>26</height>
        </rect>
       </property>
      </widget>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1147</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuImago">
    <property name="title">
     <string>Open</string>
    </property>
    <addaction name="actionOpen_Image"/>
    <addaction name="actionOpen_Video"/>
    <addaction name="actionOpen_Livestream"/>
    <addaction name="actionPause_Play"/>
   </widget>
   <widget class="QMenu" name="menuSave">
    <property name="title">
     <string>Save</string>
    </property>
    <addaction name="actionSave_Image"/>
    <addaction name="actionSave_Video"/>
   </widget>
   <addaction name="menuImago"/>
   <addaction name="menuSave"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionOpen_Image">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Image</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+I</string>
   </property>
  </action>
  <action name="actionOpen_Video">
   <property name="text">
    <string>Video</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+V</string>
   </property>
  </action>
  <action name="actionOpen_Livestream">
   <property name="text">
    <string>Livestream</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+L</string>
   </property>
  </action>
  <action name="actionPause_Play">
   <property name="text">
    <string>Pause/Play </string>
   </property>
   <property name="shortcut">
    <string>Ctrl+P</string>
   </property>
  </action>
  <action name="actionSave_Image">
   <property name="text">
    <string>Image</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSave_Video">
   <property name="text">
    <string>Video</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
